### PR TITLE
Fix Style/CaseLikeIf cop

### PR DESCRIPTION
**Pull request by archanaserver**
#### What are the changes introduced in this pull request?
Based on https://github.com/Katello/katello/pull/11009
Addresses the Style/CaseLikeIf rule by converting if-elsif statements to case-when statements

#### Considerations taken when implementing this change?
Ensured that all if-elsif statements were appropriately converted to case-when.
Make sure of existing logic and functionality of the code during the conversion.

#### What are the testing steps for this pull request?
We can run the test to ensure that no functionality is broken and all tests pass.

### CODE REVIEW

File: app/services/katello/candlepin/consumer.rb [hunk 1]
```diff
@@ -137,25 +137,26 @@ def self.distribution_to_puppet_os(name)
         return ::Operatingsystem::REDHAT_ATOMIC_HOST_OS if name == ::Operatingsystem::REDHAT_ATOMIC_HOST_DISTRO_NAME
 
         name = name.downcase
-        if name =~ /red\s*hat/
+        case name
```
Comment (m-bucher) [inline app/services/katello/candlepin/consumer.rb:140]: We could simplify this to `case name.downcase` now.

**Review by m-bucher — APPROVED** [review]
Code looks good to me.

File: app/services/katello/candlepin/consumer.rb (hunk 1 above)
Comment (archanaserver) [reply app/services/katello/candlepin/consumer.rb]: Good catch. Thanks

Comment (archanaserver) [reply app/services/katello/candlepin/consumer.rb]: @m-bucher i've updated the changes, can you take a look?

**Review by ianballou — APPROVED** [review]

**Pull request merged**